pkg/cloud/services/network/secondarycidr.go

```diff
@@ -35,7 +37,7 @@ func (s *Service) associateSecondaryCidr() error {
 		return err
 	}
 
-	if len(vpcs.Vpcs) != 1 {
+	if vpcs == nil || len(vpcs.Vpcs) != 1 {
 		return errors.Errorf("VPC not found")
 	}
```

Comment:
Since this is used in multiple places, can we extract a func out of it?

Reply:
I feel refactoring one or two lines into a separate function makes it less readable. Since, this is being used at one more place apart from this, can we keep it as it is?

Reply:
I thought refactoring it makes more sense in readability here as we can rename the func as `isVPCPresent` which makes it clear what we are trying to do with these checks, wdyt?